src/foam/nanos/crunch/ServerCrunchService.java

```diff
@@ -46,8 +46,31 @@
 {
   public static String CACHE_KEY = "CrunchService.PrerequisiteCache";
 
+  protected int cacheSequenceId_ = 0;
```

Comment:
Isn't synchronized. Either make an AtomicLong or else make two synchronized methods called incrSeq() and getSeq() (or similar) and only access this variable from them.